cadctl/cmd/investigate/investigate.go

```diff
@@ -35,12 +34,82 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var debugFlag = false
+var experimentalFlag = false
+var cadOcmClientID string
+var cadOcmClientSecret string
+var cadOcmURL string
+
 // InvestigateCmd represents the entry point for alert investigation
 var InvestigateCmd = &cobra.Command{
 	Use:          "investigate",
 	SilenceUsage: true,
 	Short:        "Filter for and investigate supported alerts",
-	RunE:         run,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		debugFlag, err := cmd.Flags().GetBool("debug")
+		if err != nil {
+			return fmt.Errorf("failed to get debug flag: %w", err)
+		}
+
+		if !debugFlag {
+			// check environment variable
+			debug, exist := os.LookupEnv("CAD_DEBUG")
+			if exist && debug == "true" {
+				debugFlag = true
+			}
+		}
+
+		if debugFlag && !ocm.HasConfigFile() {
+			return fmt.Errorf("debug flag is set, but no OCM config file found. When using debug mode the OCM config file must be set")
+		}
```

Comment:
If this has a config-file shouldn't all the other variables be retrieved from the config-file?

This would make it a bit faster, by short-circuiting the rest of the flag-handling by never even doing it with `--debug` being set + `config-file` exists.